department-of-veterans-affairs/abd-vro
# Mcp 1894 - Fix a bug and add unit test code coverage in service-data-access.

Fixing a bug and adding unit tests to increase coverage in service-data-access

## Description

Added unit tests to increase the code coverage to 80%. While adding the unit tests, a bug was detected. The bug fix is included in this PR. Also updated jwt package versions (see PR#379, #380).

### What was the problem?

The service-data-access unit test code coverage was around 65%. Added the unit tests to increase the code coverage to 80%. The added unit tests detected a bug in FieldExtractor for handling DateTimeType value, and it got fixed in this PR.

### How does this fix it?

See MCP-1894. The attachments in the Jira ticket show the code coverage increase.

### Jira Tickets

MCP-1894

- [MCP-000](https://amida.atlassian.net/browse/MCP-000)

## How to test this PR

- Step 1. Build the application.
- Step 2. Go to service-data-access Jacobo page to check the code coverage. It should show 80% as shown in the attached screen shot in MCP-1894.

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
